src/morse/index.js

```diff
@@ -58,6 +58,42 @@ const MORSE_CODE = {
 
 Object.freeze(MORSE_CODE);
 
-function morse(text) {}
+function morse(text) {
+  if (typeof text !== 'string') {
+    throw ('Please provide a morse string');
+  }
+  else {
+    let morseWords = text.trim().split('   ');
+    let morseChar = [];
+    for (let i = 0; i < morseWords.length; i++) {
+      morseChar.push(morseWords[i].split(' '));
+    }
+
+    let eachWord = [];
+    for (let i = 0; i < morseChar.length; i++) {
+      let eachChar = '';
+      for (let j = 0; j < morseChar[i].length; j++) {
+        let char = morseChar[i][j];
+        if (char in MORSE_CODE) {
+          eachChar += (MORSE_CODE[char]);
+        }
+      }
+      eachWord.push(eachChar);
+    }
+
+    let textLength = text.length;
+    let finalWord = eachWord.join(' ');
+
+    let validMorse = morseWords[0][0] in MORSE_CODE;
+
+      if (textLength === 0) {
+        return "";
+      }
+      else if (validMorse == true) {
```

Comment:
Since you return if the `textLength` is matched, you don't need the `else` here - this is also a different condition, so, you should separate it.